aldryn_people/cms_toolbar.py

```diff
@@ -7,10 +7,42 @@
 from cms.toolbar_base import CMSToolbar
 from cms.toolbar_pool import toolbar_pool
 from cms.utils.urlutils import admin_reverse
+from parler.models import TranslatableModel
 
 from .models import Group, Person
 
 
+def get_obj_from_request(model, request,
+                         pk_url_kwarg='pk',
+                         slug_url_kwarg='slug',
+                         slug_field='slug'):
+    """
+    Given a model and the request, try to extract and return an object
+    from an available 'pk' or 'slug', or return None.
+
+    Note that no checking is done that the view's kwargs really are for objects
+    matching the provided model (how would it?) so use only where appropriate.
+    """
+    language = get_language_from_request(request, check_path=True)
+    kwargs = request.resolver_match.kwargs
+    mgr = model.objects
```

Comment:
for the record, still not a fan of variable name. best to be as explicit as possible which is why I suggested model_manager